--- a/tinycloudinit.sh	2023-04-15 18:56:36.461857684 +0200
+++ b/tinycloudinit.sh	2024-03-03 14:34:22.172056974 +0100
@@ -25,6 +25,15 @@
     hostname="$(awk '/^local-hostname:/ {print $2}' $CS.rom/meta-data)"
     [ -z "$hostname" ] || hostname $hostname
   fi
+
+  if [ -f $CS.rom/lima-guestagent ]; then
+    cp $CS.rom/lima-guestagent /usr/local/bin/lima-guestagent
+    chmod 755 /usr/local/bin/lima-guestagent
+    mkdir -p /mnt/lima-cidata
+    cp $CS.rom/meta-data /mnt/lima-cidata/meta-data
+    cat $CS.rom/etc_environment >>/etc/environment
+    cp $CS.rom/lima.env /mnt/lima-cidata/lima.env
+  fi
   umount $CS.rom
 done
 rmdir $CS.rom
